src/device/mod.rs

```diff
@@ -1163,6 +1165,12 @@ impl Runtime {
     }
 
     async fn set_config(&mut self, config: &Option<Config>) -> Result {
+        if let Some(cfg) = config {
+            if cfg.this.public_key != self.get_private_key().await.map(|key| key.public())? {
+                return Err(Error::BadPublicKey);
```

Comment:
Could you also add `telio_log_err()` statement here? If we hit this case within the apps -> it will help diagnose it immediately.

Reply:
It will already be logged in `ffi/mod.rs L905` so I don't think that's necessary

Reply:
true